.github/workflows/e2e-test.yml

```
@@ -98,6 +98,19 @@ jobs:
         id: cypress-prep
         uses: ./.github/actions/prepare-cypress
 
+      - name: Download failed test artifacts
+        uses: actions/download-artifact@v4
+        continue-on-error: true
+        with:
+          pattern: failed-tests-${{ inputs.name }}-${{ inputs.edition }}
+          path: failed-tests
+
+      - name: Process failed tests
+        id: process-failed-tests
```

can we skip these steps on the first attempt? I realize we won't save much runtime, but it would make it extra clear that it only applies to re-runs
```suggestion
      - name: Process failed tests
        if: ${{ github.run_attempt != "1" }}
        id: process-failed-tests
```

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#:~:text=the%20workflow%20run.-,github.run_attempt,-string